### ISSUE TITLE
root directory renders as base module

### ISSUE THREAD

**Issue by p3trus** [sphinx-contrib/matlabdomain#81]
Am I understanding it correctly that when autodocumenting, the root directory of the matlab source code, e.g. "src", is rendered as the root module? A function "test" in src is thus rendered as "src.test". Is there an easy way to get rid of src, because it gives the wrong impression that the function must be invoked as "src.test"?

**Comment by joeced** [comment]
No, not at the moment. The autodocumenting relies on Python behavior, thus the root folder, becomes the module in which the function/classes are present. It would require at major rewrite to do otherwise. 

I'll will take into consideration in the future, but not right now.

**Comment by joeced** [comment]
@p3trus  actually, there is a fix for this. Setting `add_module_names = False` in the `conf.py`file, will remove the module name, see [add_module_names](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-add_module_names) for details. The caveat is that *all* modules names are removed, even for packages...

**Comment by joeced** [comment]
Yes, i saw what happened to packages... What would be the ideal solution for you? If you have a hierarchy of folders, they are most likely on the MATLAB path, expect for packages folders, which can be hierarchical.

So, for me, only removing the root-folder and nested normal folders would be best (or perhaps support multiple roots). Do you agree @p3trus?

**Comment by p3trus** [comment]
Let's assume i have the following folder hierarchy
```
src/global_function.m
src/+package/some_function.m
src/+package/+sub_pkg/other_function.m
```
where src is placed on the search path. Then this should be correctly documented as
```
global_function()
package.some_function()
package.sub_pkg.other_function()
```

This means the root folder must be stripped, as well as trailing + characters.

**Comment by rdzman** [comment]
I believe this should be fixed with #171.

**Comment by joeced** [comment]
Correct.  I'm slowly  but steady making the changes required for this. It's a major change of the underlying parsing and handling of matlab objects

**Comment by joeced** [comment]
Fixed in version 0.19.1
